src/main/java/com/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,75 @@
+package com.wolox.training.controllers;
+
+import com.wolox.training.dtos.UserDTO;
+import com.wolox.training.models.User;
+import com.wolox.training.services.UserService;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping
+    public ResponseEntity<List<User>> findAll() {
+        List<User> users = userService.findAll();
+        return new ResponseEntity<>(users, HttpStatus.OK);
+    }
+
+    @GetMapping("/username")
+    public ResponseEntity<User> findByUsername(@RequestParam(name = "username") String username) {
+        User user = userService.findByUsername(username);
+        return new ResponseEntity<>(user, HttpStatus.OK);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<User> findOne(@PathVariable Long id) {
+        User user = userService.findOne(id);
+        return new ResponseEntity<>(user, HttpStatus.OK);
+    }
+
+    @PostMapping
+    public ResponseEntity<User> create(@RequestBody UserDTO userDto) {
+        User user = userService.create(userDto);
+        return new ResponseEntity<>(user, HttpStatus.CREATED);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity delete(@PathVariable Long id) {
+        userService.delete(id);
+        return new ResponseEntity(HttpStatus.NO_CONTENT);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<User> updateUser(@RequestBody UserDTO userDto, @PathVariable Long id) {
+        User user = userService.updateUser(userDto, id);
+        return new ResponseEntity<>(user, HttpStatus.OK);
+    }
+
+    @PostMapping("/{id}/add")
+    public ResponseEntity<User> add(@PathVariable(name = "id") Long userId, @RequestParam Long bookId) {
+        User user = userService.addBookToAUser(userId, bookId);
+        return new ResponseEntity<>(user, HttpStatus.OK);
+    }
+
+    @PostMapping("/{id}/remove")
+    public ResponseEntity<User> create(@PathVariable(name = "id") Long userId, @RequestParam Long bookId) {
```

Comment:
same here, even a Delete Mapping could work ( /{id}/books/remove). And the 'create' method name should different 😅 like removeBook

Reply:
My bad